### PR TITLE
[Jest] migrate.ts

### PR DESCRIPTION
**Pull request by birkskyum**
Use jest-migration.md to port style-spec/migrate.test.ts

Move and rename
Run jscodeshift
Run custom codemods and lint
Manual edits
Wrap in describe

### CODE REVIEW

File: src/style-spec/migrate.test.ts [hunk 1]
```diff
@@ -107,25 +96,12 @@ describe('converts categorical function on resolvedImage type to valid expressio
...
-    }, spec.latest.$version);
+    });
     expect(migrated.layers[0].layout['icon-image']).toEqual([
         'match',
         ['get', 'type' ],
         'park',
         'some-icon',
         ''
     ]);
-    expect(validate.parsed(migrated, v8)).toEqual([]);
```
Comment (wipfli) [inline src/style-spec/migrate.test.ts]: Why remove this line?

Comment (wipfli) [reply src/style-spec/migrate.test.ts]: ...and thanks for working on this :)

File: src/style-spec/migrate.test.ts [hunk 2]
```diff
@@ -105,5 +107,6 @@ describe('migrate', () => {
             'some-icon',
             ''
         ]);
+        expect(validate(migrated, v8)).toEqual([]);
```
Comment (birkskyum) [inline src/style-spec/migrate.test.ts:110]: I have added this back.

File: src/style-spec/migrate.test.ts (hunk 1 above)
Comment (birkskyum) [reply src/style-spec/migrate.test.ts]: Looking at the bottom of this file https://github.com/maplibre/maplibre-gl-js/blob/main/src/style-spec/style-spec.ts, the .latest and .parsed seemed to just refer to the validate function. I have added a call directly to that now, as shown in the comment below.

**Review by wipfli — APPROVED** [review]

**Pull request merged**